### PR TITLE
upgrade deps and add release workflows

### PR DESCRIPTION
**Pull request by sgratzl**
Closes <list issues here>

**Prerequisites**:

* [x] Branch is up-to-date with the branch to be merged with, i.e. develop
* [x] Build is successful
* [x] Code is cleaned up and formatted 
* [x] Tested with Firefox ESR, latest Firefox, latest Chrome, Edge 18

### Summary
 
upgrade dependencies and add release workflows

### CODE REVIEW

**Comment by thinkh** [comment]
@sgratzl How does the release workflow work with your changes? How can I make a release once this PR is merged?

**Comment by sgratzl** [comment]
> @sgratzl How does the release workflow work with your changes? How can I make a release once this PR is merged?

see https://github.com/sgratzl/release-tester

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+categories:
+  - title: '🚀 Features'
+    labels:
+      - 'enhancement'
+      - 'feature'
+      - 'style'
+  - title: '🐛 Bugs Fixes'
+    labels:
+      - 'bug'
+  - title: 'Documentation'
```
Comment (thinkh) [inline .github/release-drafter.yml:12]:
```suggestion
  - title: '📖 Documentation'
```

If you want to have emojis then we can also use one for documentation. 😃

File: .github/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    labels:
+      - 'documentation'
+  - title: '🧰 Development'
```
Comment (thinkh) [inline .github/release-drafter.yml:15]:
```suggestion
  - title: '🧰 Maintenance'
```

Isn't chore not more like maintenance?

**Review by thinkh — APPROVED** [review]
Thanks, for the link to your test repo. For sake of completeness, I also add the link to the [release-drafter](https://github.com/release-drafter/release-drafter) that you use. 

I just have some minor suggestions. You can accept them if you like them and then merge this branch.

**Pull request merged**